go/common/crypto/scrape/scrape_test.go

```diff
@@ -0,0 +1,156 @@
+package scrape
+
+import (
+	"crypto/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature/signers/memory"
+)
+
+const (
+	numNodes          = 5
+	protocolThreshold = 3
+	pvssThreshold     = 3
+)
+
+func TestScrape(t *testing.T) {
+	t.Run("Basic", func(t *testing.T) { doTestBasic(t, 0, 0, 0) })
+	t.Run("MissingCommits", func(t *testing.T) { doTestBasic(t, 3, 0, 0) })
+	t.Run("MissingReveals", func(t *testing.T) { doTestBasic(t, 0, 3, 0) })
+	t.Run("MissingVerifierReveals", func(t *testing.T) { doTestBasic(t, 0, 0, 3) })
+}
+
+func doTestBasic(t *testing.T, truncateCommits, truncateReveals, truncateVerReveals int) {
```

Comment:
One useful thing to test would be serialization round-trips of these various types to make sure there are no surprises.

Reply:
Yeah I just came to the realization that types that contain `kyber.Scalar`/`kyber.Point` don't handle s11n well because they're interface types.

Reply:
Fixed and now tested.